### PR TITLE
feat/add authentication

### PR DESCRIPTION
**Pull request by RahulPrajapati221**
Add authentication and middleware

### CODE REVIEW

**Comment by RahulPrajapati221** [comment]
@neelg-webelight please check

File: src/models/User.js [hunk 1]
```diff
@@ -69,16 +69,6 @@ userSchema.methods.toJSON = function(){
     return userObject
 }
 
-
```
Comment (neelg-webelight) [inline src/models/User.js]: Remove unused import and change file name to user-model.js and move to user module

File: src/modules/tasks/task-services.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    await task.save();
+    return task;
+  }
+  
+export const delTask= async(taskId, ownerId) =>{
+    const task = await Task.findOneAndDelete({
+        _id:taskId,
+        owner_id:ownerId
+    });
+    return task;
```
Comment (neelg-webelight) [inline src/modules/tasks/task-services.js:64]: use meaningful variable names

File: src/modules/tasks/taskConroller.js [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+export const getTasks = async (req, resp) => {
+  const { match, limit, skip, sort } = getT(req.query);
+  try {
+    await req.user.populate({
+      path: "tasks",
+      match,
+      options: {
+        limit: limit,
+        skip: skip,
```
Comment (neelg-webelight) [inline src/modules/tasks/taskConroller.js:41]: use shortened syntax.

File: src/modules/tasks/taskConroller.js [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+
+const {
+  emailError,
+  passError,
+  ageError,
+  badRequest,
+  serverError,
+  unauthorized,
+  notFound,
+} = errorMess;
```
Comment (neelg-webelight) [inline src/modules/tasks/taskConroller.js:31]: Remove unused import

File: src/modules/tasks/task-services.js [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+
+export function validTask(updates) {
+    const allowedUpdates = ["description", "completed"];
+
+    const isValidOperation = updates.every((update) => {
+      return allowedUpdates.includes(update);
+    });
+    return isValidOperation;
+  }
```
Comment (neelg-webelight) [inline src/modules/tasks/task-services.js:49]: Validation function should be common for task and user module. move it into utils.

File: src/modules/tasks/taskConroller.js [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+
+export const getTasks = async (req, resp) => {
+  const { match, limit, skip, sort } = getT(req.query);
+  try {
+    await req.user.populate({
```
Comment (neelg-webelight) [inline src/modules/tasks/taskConroller.js:36]: store req.user.task into variable. send variable in the response

File: src/modules/users/user-services.js [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+export const createUser = async (reqBody) =>{
+  const user = await User.create(reqBody);
+  const token = await generateToken(user);
+  return { user, token };
+};
+
+export const userLogin=(user) =>{
+  const token = generateToken(user);
+  return token;
+}
```
Comment (neelg-webelight) [inline src/modules/users/user-services.js:13]: use directly generateToken function where token is needed.

File: src/constant.js [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+  notFound: (value) => `${value} Not found, Please try again.`,
+};
+
+export const taskMsgs = {
+  taskNotfound: "Task",
+};
+
+export const userMsgs = {
+  userNotfound: "User",
+};
```
Comment (neelg-webelight) [inline src/constant.js:34]: Use sore task and user both in single object.
export const constants = {
  user: "User",
  task: "Task"
}

File: src/modules/tasks/taskConroller.js [hunk 9]
```diff
@@ -0,0 +1,114 @@
...
+  }
+};
+
+//Update Task
+export const updateTask = async (req, resp) => {
+  const updates = Object.keys(req.body);
+  const allowedUpdates = ["description", "completed"];
+  const updateField = { updates, allowedUpdates };
+
+  const isValidOperation = validUpdate(updateField);
```
Comment (neelg-webelight) [inline src/modules/tasks/taskConroller.js:83]: No need to create updateField variable.
Pass arguments directly into functions.

File: src/modules/users/userController.js [hunk 10]
```diff
@@ -0,0 +1,114 @@
...
+  }
+};
+
+// update user
+export const updateUser = async (req, resp) => {
+  const updates = Object.keys(req.body);
+  const allowedUpdates = ["name", "email", "password", "age"];
+  const updateField = { updates, allowedUpdates };
+
+  const isValidOperation = validUpdate(updateField);
```
Comment (neelg-webelight) [inline src/modules/users/userController.js:81]: No need to create updateField variable.
Pass arguments directly into functions.

**Pull request merged**